### PR TITLE
big-O notation: parenthesis for function calls, explicit multiplication

### PR DESCRIPTION
**Pull request by RalfJung**
I saw `O(n m log n)` in the docs and found that really hard to parse. In particular, I don't think we should use blank space as syntax for *both* multiplication and function calls, that is just confusing.

This PR makes both multiplication and function calls explicit using Rust-like syntax. If you prefer, I can also leave one of them implicit, but I believe explicit is better here.

While I was at it I also added backticks consistently.

### CODE REVIEW

File: src/liballoc/collections/binary_heap.rs [hunk 1]
```diff
@@ -644,7 +643,7 @@ impl<T: Ord> BinaryHeap<T> {
     /// The remaining elements will be removed on drop in heap order.
     ///
     /// Note:
-    /// * `.drain_sorted()` is O(n lg n); much slower than `.drain()`.
+    /// * `.drain_sorted()` is `O(n * lg(n))`; much slower than `.drain()`.
```
Comment (RalfJung) [inline src/liballoc/collections/binary_heap.rs:646]: I don't know what `lg` is supposed to mean here. Is it `log` with a typo, or is it supposed to indicate a particular base?

File: src/liballoc/collections/btree/map.rs [hunk 2]
```diff
@@ -40,7 +40,7 @@ use UnderflowResult::*;
 /// performance on *small* nodes of elements which are cheap to compare. However in the future we
 /// would like to further explore choosing the optimal search strategy based on the choice of B,
 /// and possibly other factors. Using linear search, searching for a random element is expected
-/// to take O(B log<sub>B</sub>n) comparisons, which is generally worse than a BST. In practice,
+/// to take O(B * log<sub>B</sub>(n)) comparisons, which is generally worse than a BST. In practice,
```
Comment (RalfJung) [inline src/liballoc/collections/btree/map.rs:43]: Elsewhere we are using `log_2` to indicate the base, so this is inconsistent. But for now I decided not to change this.

File: src/liballoc/collections/binary_heap.rs (hunk 1 above)
Comment (Dylan-DPC) [reply src/liballoc/collections/binary_heap.rs:646]: this is the only place it is used here, i assume it is a typo

Comment (shepmaster) [reply src/liballoc/collections/binary_heap.rs:646]: https://www.math10.com/en/algebra/logarithm-log-ln-lg.html suggests that `lg` == log base 10

https://mathworld.wolfram.com/Lg.html suggests it might mean log base 2

Comment (RalfJung) [reply src/liballoc/collections/binary_heap.rs:646]: I mean anyway the difference between base 2 and base 10 is a constant, so in the context of big-O the base just doesn't matter... but then there are two places in the docs that explicitly give a base in big-O.

Comment (RalfJung) [reply src/liballoc/collections/binary_heap.rs]: So, I changed this here to `log` and removed the two explicitly given bases.

**Comment by shepmaster** [comment]
Thanks!
